sdk/python/smithy_python/components/component.py

```
@@ -0,0 +1,132 @@
+import uuid
+from smithy_python.helpers.logger import log
+from typing import Union, List, Optional
+from logging import Logger
+from smithy_python.remote_store.findings_service.v1 import findings_service_pb2
+from smithy_python.dbmanagers import RemoteDBManager, PostgresDBManager, SqliteDBManager
+from abc import ABC
+from smithy_python.enums.db_type_enum import DBTypeEnum
+
+
+class Component(ABC):
+    """
+    A component in the Smithy Python SDK that represents a component in the smithy framework.
+    """
+
+    def __init__(
+        self,
+        instance_id: Union[uuid.UUID, str],
+        db_type: DBTypeEnum,
+        logger: Optional[Logger] = None,
+    ) -> None:
+        """
+        Initializes a new instance of the Component class.
+        **ATTENTION**: Currently only the `remote` database type is supported, all other types will raise a `NotImplementedError`.
+        :param instance_id: The UUID of the instance (aka the Workflow run to be analyzed)
+        :type instance_id: Union[uuid.UUID, str]
+        :param db_type: The database mode for the enricher, DBTypeEnum, can be either `SQLITE`, `POSTGRES` or `REMOTE`. (remote is gRPC)
+        :type db_type: DBTypeEnum
+        :param logger: Optional an instance of the Logger class for logging, if not provided, a default logger will be used.
+        :type logger: Optional[Logger]
+        :raises ValueError: If the instance_id is not a valid UUID.
+        :raises TypeError: If the instance_id is not a string or UUID object, or the logger is not an instance of Logger or None.
+        """
+
+        if not logger:
+            self.log = log
+        elif isinstance(logger, Logger):
+            self.log = logger
+        else:
+            raise TypeError("logger must be an instance of Logger or None.")
+
+        if instance_id is None or (
```

if the instance_id is a string we should first try to parse it as a uuid4 and if that fails we should raise an exception. the `instance_id` should be a uuid4 field in the class. we can merge the `_validate_uuid` into this part of the constructor. also we should check that it's a UUID4, the Python API doesn't offer a specific identifier so we should do our own checks.

---

https://github.com/smithy-security/smithy/pull/1016/commits/2d6dd386defa5ba4d664bfba450702486452de38